domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -97,15 +336,62 @@ class ProfileManagementController @Inject constructor(
   }
 
-  /**
-   * Sets the currentProfileId to the selected profile in ProfileChooserFragment.
-   * Checks to ensure that profileId corresponds to an existing profile.
-   *
-   * @param profileId the ID corresponding to the profile being set.
-   * @return a [LiveData] that indicates the success/failure of this set operation.
-   */
-  fun setCurrentProfileId(profileId: ProfileId): LiveData<AsyncResult<Any?>> {
-    currentProfileId = profileId.internalId
-    return MutableLiveData(AsyncResult.success<Any?>(null))
+  private fun isUniqueName(newName: String, profileDatabase: ProfileDatabase): Boolean {
+    val lowerCaseNewName = newName.toLowerCase()
+    profileDatabase.profilesMap.values.forEach {
+      if (it.name.toLowerCase(Locale.getDefault()) == lowerCaseNewName) {
+        return false
+      }
+    }
+    return true
+  }
+
+  private fun saveImageToInternalStorage(avatarImagePath: Uri, profileDir: File): String? {
+    val bitmap = MediaStore.Images.Media.getBitmap(context.contentResolver, avatarImagePath)
+    val imageFile = File(profileDir, "profile.png")
+    try {
+      FileOutputStream(imageFile).use { fos ->
+        rotateAndCompressBitmap(avatarImagePath, bitmap, 300)
+          .compress(Bitmap.CompressFormat.PNG, /* quality= */ 100, fos)
+      }
+    } catch (e: Exception) {
+      logger.e("ProfileManagementController", "Failed to store user submitted avatar image", e)
+      return null
+    }
+    return imageFile.absolutePath
+  }
+
+  private fun onlyLetters(name: String): Boolean {
+    return name.matches(Regex("^[ A-Za-z]+\$"))
+  }
+
+  // https://stackoverflow.com/questions/3934331/how-to-hash-a-string-in-android
+  private fun md5(s: String): String {
+    try {
+      val digest = MessageDigest.getInstance("MD5")
+      digest.update(s.toByteArray(Charset.forName("US-ASCII")), 0, s.length)
+      val magnitude = digest.digest()
+      val bi = BigInteger(1, magnitude)
+      val hash = String.format("%0" + (magnitude.size.shl(1)) + "x", bi)
+      return hash
+    } catch (e: NoSuchAlgorithmException) {
+      logger.e("ProfileManagementController", "No such algorithm when creating md5 hash for gravatar", e)
+    }
+    return ""
+  }
+
+  private fun rotateAndCompressBitmap(uri: Uri, bitmap: Bitmap, cropSize: Int): Bitmap {
+    val croppedBitmap = ThumbnailUtils.extractThumbnail(bitmap, cropSize, cropSize)
+    val inputStream = context.contentResolver.openInputStream(uri)!!
+    val orientation = ExifInterface(inputStream).getAttributeInt(ExifInterface.TAG_ORIENTATION, 1)
+    var rotate = 0
+    when (orientation) {
+      ExifInterface.ORIENTATION_ROTATE_90 -> rotate = 90
+      ExifInterface.ORIENTATION_ROTATE_180 -> rotate = 180
+      ExifInterface.ORIENTATION_ROTATE_270 -> rotate = 270
+    }
+    val matrix = Matrix()
+    matrix.postRotate(rotate.toFloat())
+    return Bitmap.createBitmap(croppedBitmap, /* x= */0, /* y= */ 0, cropSize,  cropSize, matrix, /* filter= */true)
```

Comment:
Nit: space after ``*/`` in a couple places.

Reply:
Fixed.